src/app/container/add-tag/add-tag.component.ts

```diff
@@ -179,22 +179,20 @@ export class AddTagComponent implements OnInit, AfterViewChecked {
       }
     }
   }
+
   // Validation ends here
   // Checks if the currently edited test parameter file already exists
   // TODO: This code is repeated in version-modal.component.ts for tools, move it somewhere common
-  hasDuplicateTestJson(type) {
-    if (type === 'cwl') {
-      if (this.unsavedCWLTestParameterFilePaths.includes(this.unsavedTestCWLFile)) {
-        return true;
-      } else {
-        return false;
-      }
-    } else if (type === 'wdl') {
-      if (this.unsavedWDLTestParameterFilePaths.includes(this.unsavedTestWDLFile)) {
-        return true;
-      } else {
-        return false;
-      }
+  // TODO: This is also executed a bajillion times
+  hasDuplicateTestJson(type: ToolDescriptor.TypeEnum) {
+    if (type === this.DescriptorType.CWL) {
+      return this.unsavedWDLTestParameterFilePaths.includes(this.unsavedTestCWLFile) ||
+        this.unsavedCWLTestParameterFilePaths.includes(this.unsavedTestCWLFile) ||
+        (this.unsavedTestCWLFile === this.unsavedTestWDLFile && this.unsavedTestCWLFile);
+    } else if (type === this.DescriptorType.WDL) {
```

Comment:
these predicates are quite complex, could use a comment at least
potential for simplification possibly (is this the same as merging unsavedWDLTestParameterFilePaths and unsavedWDLTestParameterFilePaths and checking whether the merge contains the unsaved test file? not sure what is going on in the last bracket 

so: 
minimum: code comment what each branch is doing and what is going on in the last bracket of each branch
stretch goal: simplify this